packages/babel-cli/src/babel/util.js

```diff
@@ -81,3 +83,48 @@ export function requireChokidar() {
     throw err;
   }
 }
+
+export function getSettings(filenames) {
+  let File = require("babel-core").File;
+  let generalOptions = new File( { filename: "unknown" } ).opts;
+  let allOptions = [];
+
+  allOptions.push(generalOptions);
+
+  each(filenames, function (file) {
+    let fileOptions = new File( { filename: file } ).opts,
+        thisFileOptions = {};
+
+    each(fileOptions, function (fileOption, key) {
+      if (!generalOptions.hasOwnProperty(key) || JSON.stringify(generalOptions[key]) != JSON.stringify(fileOption)) {
+        thisFileOptions[key] = fileOption;
+      }
+    });
+    allOptions.push(thisFileOptions);
+  });
+
+  let printObject = function (filename, opts) {
+    console.log(`--- ${filename} options ---`);
+    each(opts, function (option, key) {
+      console.log(key, ": ", option);
+    });
+    console.log();
+  };
+
+  process.stdout.write(`node version: `);
+  child.execSync("node -v", {stdio:[0, 1]});
+  process.stdout.write(`npm version: `);
+  child.execSync("npm -v", {stdio:[0, 1]});
+  process.stdout.write(`packages:\n`);
+  child.execSync("npm list", {stdio:[0, 1]});
```

Comment:
might be useful to filter out non-babel packages for `npm list` -> `npm list | grep babel` but can't assume `grep` is available